x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/validation/SourceDestValidator.java

```diff
@@ -65,6 +65,7 @@
     private final RemoteClusterLicenseChecker remoteClusterLicenseChecker;
     private final String nodeName;
     private final String license;
+    private final boolean includeDatastreams;
```

Comment:
makes me wonder, if we have a use case, where we do not want data streams, in this PR it seems not.

Except the unit tests: Do they fail for `includeDatastreams=true`?

If there is no good reason, I suggest to keep it simple and make data streams default.